linchpin/provision/roles/azure/files/schema.json

```diff
@@ -0,0 +1,45 @@
+{
+    "res_defs": {
+        "type": "list",
+        "schema": {
+            "anyof": [
+                {
+                    "type": "dict",
+                    "schema": {
+                        "role": {
+                            "type": "string",
+                            "required": true,
+                            "allowed": [
+                                "azure_vm"
+                            ]
+                        },
+                        "vm_name": {
+                            "type": "string",
+                            "required": true
+                        },
+                        "image": {
+                            "type": "string",
+                            "required": true
+                        },
+                        "vm_username": {
+                            "type": "string",
+                            "required": true
```

Comment:
Are all of these fields really required or are there some for which we can set reasonable defaults?  e.g. {{ ansible_ssh_user }} for vm_username

Reply:
yes, will do